torch/jit/_recursive.py

```diff
@@ -127,6 +127,12 @@ def infer_type(name, item):
 
         added_names.add(name)
 
+    for idx, fn in nn_module._forward_hooks.items():
+        concrete_type_builder.add_forward_hook(fn.__name__)
```

Comment:
This should use `_jit_internal._qualified_name` instead of `fn.__name__` so the naming is consistent for all objects

Reply:
After discussing offline, we decided to leave this part of the code as is since we don't have a good way of parsing qualified names on the import side.